### PR TITLE
(+) Add ISupportPackageContentProvider to provide extensibility at support package creation

### PR DESCRIPTION
**Pull request by alexfdezsauco**
Fixes # .

### Checklist

- [X] I have included examples or tests
- [X] I have cleaned up the commit history (use rebase and squash)

### Changes proposed in this pull request:

- Add ISupportPackageContentProvider to provide extensibility at support package creation

### CODE REVIEW

File: src/Orc.SupportPackage.Example/SupportPackage/Providers/CustomSupportPackageContentProvider.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+{
+    using System.Collections.Generic;
+    using System.IO;
+
+    public sealed class CustomSupportPackageContentProvider : ISupportPackageContentProvider
+    {
+        #region Methods
+        public IEnumerable<SupportPackageFileSystemArtifact> GetSupportPackageFileSystemArtifacts()
+        {
+            var demoProjectPath = @"c:\Tmp\Orc.SupportPackage.Example";
```
Comment (GeertvanHorrik) [inline src/Orc.SupportPackage.Example/SupportPackage/Providers/CustomSupportPackageContentProvider.cs]: Can't we use Environment.WorkingDirectory or GetType().GetAssemblyEx().GetDirectory() instead?

File: src/Orc.SupportPackage.Xaml/Orc.SupportPackage.Xaml.Shared/Models/SupportPackageDirectory.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace Orc.SupportPackage
+{
+    using Catel;
+    using Catel.Fody;
+
+    public class SupportPackageDirectory : SupportPackageFileSystemArtifact
+    {
+        #region Constructors
+        public SupportPackageDirectory(string title, string directoryName, bool includeInSupportPackage = true):base(title, includeInSupportPackage)
```
Comment (GeertvanHorrik) [inline src/Orc.SupportPackage.Xaml/Orc.SupportPackage.Xaml.Shared/Models/SupportPackageDirectory.cs]: split : base on new line

File: src/Orc.SupportPackage.Xaml/Orc.SupportPackage.Xaml.Shared/Models/SupportPackageFileNamePattern.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+namespace Orc.SupportPackage
+{
+    using System.Linq;
+    using Catel;
+    using Catel.Fody;
+
+    public class SupportPackageFileNamePattern : SupportPackageFileSystemArtifact
+    {
+        #region Constructors
+        public SupportPackageFileNamePattern(string title, string[] fileNamePatterns, bool includeInSupportPackage = true):base((title + " " + fileNamePatterns.Aggregate("(", (current, fileExtension) => current + fileExtension + " | ").TrimEnd('|', ' ') + ")").Trim(), includeInSupportPackage)
```
Comment (GeertvanHorrik) [inline src/Orc.SupportPackage.Xaml/Orc.SupportPackage.Xaml.Shared/Models/SupportPackageFileNamePattern.cs]: split base

File: src/Orc.SupportPackage.Xaml/Orc.SupportPackage.Xaml.Shared/ViewModels/SupportPackageViewModel.cs [hunk 4]
```diff
@@ -55,36 +59,19 @@ public class SupportPackageViewModel : ViewModelBase
             CreateSupportPackage = new TaskCommand(OnCreateSupportPackageExecuteAsync, OnCreateSupportPackageCanExecute);
             OpenDirectory = new Command(OnOpenDirectoryExecute, OnOpenDirectoryCanExecute);
 
-            SupportPackageFileTypes = new List<SupportPackageFileType>
+            SupportPackageFileSystemArtifacts = new List<SupportPackageFileSystemArtifact>();
```
Comment (GeertvanHorrik) [inline src/Orc.SupportPackage.Xaml/Orc.SupportPackage.Xaml.Shared/ViewModels/SupportPackageViewModel.cs]: A bit more logging would be nice here. E.g. for each package content provider, log the type + tell that we are about to retrieve the artifacts

File: src/Orc.SupportPackage/Orc.SupportPackage.Shared/Services/Interfaces/ISupportPackageService.cs [hunk 5]
```diff
@@ -7,13 +7,14 @@
...
-    using System;
     using System.Threading.Tasks;
 
     public interface ISupportPackageService
     {
         Task<bool> CreateSupportPackageAsync(string zipFileName);
 
         Task<bool> CreateSupportPackageAsync(string zipFileName, string[] excludeFileNamePatterns);
+
+        Task<bool> CreateSupportPackageAsync(string zipFileName, string[] directories, string[] excludeFileNamePatterns);
```
Comment (GeertvanHorrik) [inline src/Orc.SupportPackage/Orc.SupportPackage.Shared/Services/Interfaces/ISupportPackageService.cs]: Mark others obsolete?

**Comment by alexfdezsauco** [comment]
Fixed

**Pull request merged**